src/proj_equirect.jl

```diff
@@ -197,26 +189,116 @@ function BlockDiagEquiRect{B}(vector_of_blocks::Vector{A}, proj::P) where {B<:Ba
     BlockDiagEquiRect{B}(block_matrix, proj)
 end
 
-# ## Linear Algebra basics 
+# ## Linear Algebra: tullio accelerated (operator, field)
 
-*(M::BlockDiagEquiRect{B}, f::EquiRectField) where {B<:Basis} = M * B(f)
+# M * f
 
-function *(M::BlockDiagEquiRect{AzFourier}, f::EquiRectAzFourier)
+Base.:*(M::BlockDiagEquiRect{B}, f::EquiRectField) where {B<:Basis} = M * B(f)
+
+function Base.:*(M::BlockDiagEquiRect{AzFourier}, f::EquiRectAzFourier)
     promote_metadata_strict(M.proj, f.proj) # ensure same projection
-    EquiRectAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.metadata)
+    EquiRectAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.proj)
 end
 
-function *(M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
+function Base.:*(M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
     promote_metadata_strict(M.proj, f.proj) # ensure same projection
-    EquiRectQUAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.metadata)
+    EquiRectQUAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.proj)
+end
```

Comment:
You can do both bases at once with:

```julia
function Base.:*(M::BlockDiagEquiRect{B}, f::F) where {B, F<:EquiRectField{B}}
    promote_metadata_strict(M.proj, f.proj) # ensure same projection
    F(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.proj)
end
```

Similarly for the other functions.

Reply:
Great. Thanks! Been intending to do that but couldn't remember `EquiRectField{B}` and always felt a bit too lazy to track it down. I'll clean it up soon